ixdiagnose/plugins/vm.py

```diff
@@ -1,10 +1,23 @@
-from ixdiagnose.utils.middleware import MiddlewareCommand
+import typing
+
+from ixdiagnose.utils.formatter import remove_keys
+from ixdiagnose.utils.middleware import MiddlewareClient, MiddlewareCommand
 
 from .base import Plugin
-from .metrics import FileMetric, MiddlewareClientMetric
+from .metrics import FileMetric, MiddlewareClientMetric, PythonMetric
 from .prerequisites import VMPrerequisite
 
 
+def vm_query(client: MiddlewareClient, context: typing.Any) -> list:
+    query_output = client.call('vm.query')
+    for item in query_output:
```

Comment:
This for loop can be simplified and sped up at the same time. I was thinking something like:
```
for i in query_output:
    for j in i.get('devices', []):
        try:
            j['attributes']['password'] = 'REDACTED'
        except KeyError:
            continue